File: drift/src/main/java/drift/com/drift/helpers/LogoutHelper.java
package drift.com.drift.helpers;

import drift.com.drift.managers.AttachmentManager;
import drift.com.drift.managers.ConversationManager;
import drift.com.drift.managers.MessageManager;
import drift.com.drift.managers.PresentationManager;
import drift.com.drift.managers.SocketManager;
import drift.com.drift.managers.UserManager;
import drift.com.drift.model.Auth;
import drift.com.drift.model.Embed;

public class LogoutHelper {
    public static void logout(){
        PresentationManager.getInstance().closePopupView();
        UserManager.getInstance().clearCache();
        ConversationManager.getInstance().clearCache();
        MessageManager.getInstance().clearCache();
        AttachmentManager.getInstance().clearCache();
        Auth.deleteAuth();
        SocketManager.getInstance().disconnect();
    }
}

File: drift/src/main/java/drift/com/drift/helpers/LogoutHelper.kt
package drift.com.drift.helpers

import drift.com.drift.managers.AttachmentManager
import drift.com.drift.managers.ConversationManager
import drift.com.drift.managers.MessageManager
import drift.com.drift.managers.PresentationManager
import drift.com.drift.managers.SocketManager
import drift.com.drift.managers.UserManager
import drift.com.drift.model.Auth

object LogoutHelper {
    fun logout() {
        PresentationManager.closePopupView()
        UserManager.clearCache()
        ConversationManager.clearCache()
        MessageManager.clearCache()
        AttachmentManager.clearCache()
        Auth.deleteAuth()
        SocketManager.disconnect()
    }
}
